RecoTracker/PixelVertexFinding/plugins/alpaka/fitVertices.h

```diff
@@ -74,7 +74,19 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE::vertexFinder {
     alpaka::syncBlockThreads(acc);
     // reuse nn
     for (auto i : cms::alpakatools::uniform_elements(acc, foundClusters)) {
-      ALPAKA_ASSERT_ACC(wv[i] > 0.f);
+      bool const wv_cond = (wv[i] > 0.f);
+      if (not wv_cond) {
+        printf("ERROR: wv[%d] (%f) > 0.f failed\n", i, wv[i]);
```

Comment:
I've seen that addition of a printf requires a lot of registers.
How much slower is the code after this change, is it significant?

Reply:
Is it relevant? The printout should occur every other ten of millions events?

Reply:
as I recall, if it is compiled, that's enough

Reply:
I don't have any feeling (and neither a measurement). If @cms-sw/heterogeneous-l2 have a feeling about it please chime in, otherwise it will have to be measured.

Reply:
Yes, it's relevant, because the increase in register use is there even if the branch is never taken at runtime.

In general I would prefer not to add any `printf` - but since we have been unable to reproduce the assertion offline, I think in this case the extra resource usage is warranted.

Assuming this helps figuring out the source of the problem and fixing it, afterwards we can move the `printf` inside some `if constexpr (debug) { ... }` clause.

Reply:
> How much slower is the code after this change, is it significant?

Hard to guess, it would need to be measured 🤷🏻‍♂️

Reply:
We checked the HLT throughput (details in [*]).

 - Pre-PR: 618.8 +/- 1.3 events/sec ([json](https://github.com/user-attachments/files/16503412/out_240805_test_cmssw45631_54ca188_prePR-CMSSW_14_0_13_patch1_MULTIARCHS-00-gpu_mps.json)).
 - Post-PR: 616.1 +/- 1.0 events/sec ([json](https://github.com/user-attachments/files/16503417/out_240805_test_cmssw45631_54ca188_posPR-CMSSW_14_0_13_patch1_MULTIARCHS-00-gpu_mps.json)).

The impact of this PR on the HLT throughput is well below 1%, and somewhat within the uncertainties of these estimates.

[*]
 - HLT menu: `/cdaq/physics/Run2024/2e34/v1.4.3/HLT/V2` (current pp online menu).
 - Release: `CMSSW_14_0_13_patch1_MULTIARCHS` (with and without this PR).
 - Node: `hilton-c2b01-44-01` (same hardware as a 2022/23 HLT node) (CPUs: 2 AMD EPYC 7763 64-Core; GPUs: 2 NVIDIA Tesla T4).
 - 8 jobs, 32 threads and 24 streams per job.
 - NVIDIA MPS enabled.
 - Each of the two measurements reported above is the average of 4 repetitions.